### DISCUSSION TITLE
Compatibility PHP 8.2.9 / Laravel 10.

### DISCUSSION THREAD

**Discussion by OctavioBernalGH** [yajra/laravel-oci8#810] [Q&A] [answered]
Good morning, we have problems installing the package, our build is based on PHP 8.2.9 and Laravel 10, is it compatible with these versions? We need to be able to connect/create connections to oracle database from the database.php file.

Is there any guide for these versions ?

**Comment by yajra** [comment]
Currently using 8.2.11 and Laravel 10 and works well.

On ENV you can set:

```php
DB_CONNECTION=oracle
DB_HOST=127.0.0.1
DB_DATABASE=xe
DB_USERNAME=forge
DB_PASSWORD=forge
```

Or publish the `oracle.php` config and update/add entries accordingly.

**Reply by OctavioBernalGH** [reply]
And did you install oci8 ? Standard oci8 does not work with php 8.2.9 seems. I have problems at use oracle connections.

How i can solve it ?

**Reply by yajra** [reply]
Yes, you need to have oci8 extension installed.

**Reply by OctavioBernalGH** [reply]
I have that error:

Warning: PHP Startup: Unable to load dynamic library 'oci8' (tried: C:/laragon/bin/php/php-8.2.9-Win32-vs16-x64/ext\oci8 (No se puede encontrar el módulo especificado), C:/laragon/bin/php/php-8.2.9-Win32-vs16-x64/ext\php_oci8.dll (No se encontró el proceso especificado)) in Unknown on line 0

I have php_oci8.dll in ext folder and enabled in the php.ini

**Reply by OctavioBernalGH** [reply]
And i have the next errors:

PS C:\laragon\www\oracle-app> composer require yajra/laravel-oci8:^9.0

Warning: PHP Startup: Unable to load dynamic library 'oci8' (tried: C:/laragon/bin/php/php-8.2.9-Win32-vs16-x64/ext\oci8 (No se puede encontrar el m├│dulo especificado), C:/laragon/bin/php/php-8.2.9-Win32-vs16-x64/ext\php_oci8.dll (No se encontr├│ el proceso especificado)) in Unknown on line 0
The "9.0" constraint for "yajra/laravel-oci8" appears too strict and will likely not match what you want. See https://getcomposer.org/constraints
./composer.json has been updated
Running composer update yajra/laravel-oci8
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires yajra/laravel-oci8 9.0 -> satisfiable by yajra/laravel-oci8[v9.0.0].
    - yajra/laravel-oci8 v9.0.0 requires ext-oci8 >=3.0.1 -> it is missing from your system. Install or enable PHP's oci8 extension.

To enable extensions, verify that they are enabled in your .ini files:
    - C:\laragon\bin\php\php-8.2.9-Win32-vs16-x64\php.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-oci8` to temporarily ignore these required extensions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Reply by yajra** [reply]
Haven't used Windows for years so can't help with fixing your setup. However, installing the OCI8 extension correctly should fix your issue.

**Reply by OctavioBernalGH** [reply]
I will try to solve it and see if I can.

**Reply by OctavioBernalGH** [reply, marked as answer]
@yajra 
I created a new project using version 8.2.11 and laravel 10.X, configured the database and enabled the php extension oci8_19 and it worked perfectly.

**Reply by yajra** [reply]
Glad you were able to make it work, thanks for the updates.
